index.js

```diff
@@ -90,6 +90,33 @@ var Client = redefine.Class({
 
   space: function(callback) {
     return nodeify(this._request(''), callback);
+  },
+
+  _pagedSync: function (sync) {
+    var self = this;
+    return this._request('/sync', sync.query)
+               .then(function (data) {
+                 sync.append(data);
+                 if(!sync.done){
+                   return self._pagedSync(sync);
+                 } else {
+                   return {
+                     items: sync.items,
+                     nextSyncToken: sync.nextSyncToken
+                   };
+                 }
+               });
+  },
+
+  sync: function (object, callback) {
+    var query = Query.parse(object);
+    var items = [];
```

Comment:
Ah good catch. I wrote an initial version of the code that was very different, then refactored stuff into the Sync class and this is somehow a leftover.